CONTRIBUTING.md

```diff
@@ -380,29 +385,25 @@ To add a new supported tool:
-   5. Add it to the list of tools shown by `terra app list` by adding the new command class to
-   the list of sub-commands in the `@Command` annotation of the `Main.class`. This means you can
-   invoke the command by prefixing it with terra (e.g. `terra dsub -version`).
+      (e.g. `terra app execute dsub --version`). This command just runs the Docker container and executes the command,
+      without requiring any new Java code. This `terra app execute` command is intended for debugging only; this won't
+      be how users call the tool.
+   4. Add a new command class in the `src/main/java/bio/terra/cli/command/app/passthrough` package. Copy/paste an
+      existing class in that same package as a starting point.
+   5. Add it to the list of tools shown by `terra app list` by adding the new command class to the list of sub-commands
+      in the `@Command` annotation of the `Main.class`. This means you can invoke the command by prefixing it with
+      terra (e.g. `terra dsub -version`).
    6. When you run e.g. `terra dsub -version`, the CLI:
-      - Launches a Docker container
-      - Runs the `terra_init.sh` script in the `docker/scripts` directory, which activates the user’s
-      pet service account and sets the workspace project
-      - Runs the `dsub` command
-   7. You can pass environment variables through to the Docker container by populating a `Map` and
-   passing it to the `DockerAppsRunner.runToolCommand` method. Two environment variables are always
-   passed:
-       - `GOOGLE_CLOUD_PROJECT` = the workspace project id
-       - `GOOGLE_APPLICATION_CREDENTIALS` = the pet service account key file
-   8.  You can mount directories on the host machine to the Docker container by populating a second
-   `Map` and passing it to the same `DockerAppsRunner.runToolCommand` method. The current working
-   directory is always mounted to the Docker container.
+       - [Docker mode only] Launches a Docker container
+       - [Docker mode only] Runs the `terra_init.sh` script in the `docker/scripts` directory, which sets the workspace project
+       - Runs the `dsub` command
+   7. Environment variables that are always set: `GOOGLE_SERVICE_ACCOUNT_EMAIL`, `GOOGLE_CLOUD_PROJECT`, `TERRA_<resource_name>`  
+      If you need additional environment variables, you can pass them into `CommandRunner.runToolCommand()`.
+   8. You can mount directories on the host machine to the Docker container by populating a second
+         `Map` and passing it to the same `DockerAppsRunner.runToolCommand` method. The current working directory is always
+         mounted to the Docker container.
    9. Publish the new Docker image and update the default image that the CLI uses to the new version
-   (see instructions in section above).
+         (see instructions in section above).
 
 #### Commands
 The `command` package contains the hierarchy of the commands as they appear to the user.
```

Comment:
Below on line 563, it says "Tests only run in docker mode." That's partially true today, and will become less so soon. I'd document that when we setup tests we pull the docker container, and Docker is present in the test environment, so tests are installed in Docker mode (if that's not more confusing). They're not necessarily _run_ in Docker mode, as some of them switch `app-launch` to `LOCAL_PROCESS`.

Reply:
Changed to `Tests run in docker mode by default` and linked to where I think that is set.

(I don't think `‎‎tools/install.sh` runs for unit tests. Unit tests run the Java commands directly.)